core/embed/rust/src/ui/layout_caesar/component_msg_obj.rs

```diff
@@ -117,17 +117,25 @@ impl ComponentMsgObj for CoinJoinProgress {
 
 impl ComponentMsgObj for NumberInput {
     fn msg_try_into_obj(&self, msg: Self::Msg) -> Result<Obj, Error> {
-        (CONFIRMED.as_obj(), msg.try_into()?).try_into()
+        match msg {
+            Self::Msg::Cancel => (CANCELLED.as_obj(), 0.try_into()?).try_into(),
+            Self::Msg::Choice { item, .. } => (CONFIRMED.as_obj(), item.try_into()?).try_into(),
```

Comment:
while we're here: i forget, why do we return a tuple instead of `int / Cancelled` ?